### PR TITLE
Revert "[SPARK-30625][SQL] Support `escape` as third parameter of the `like` function

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?

In the PR, I propose to revert the commit 8aebc80e0e67bcb1aa300b8c8b1a209159237632.

### Why are the changes needed?
See the concerns https://github.com/apache/spark/pull/27355#issuecomment-584344438

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By existing test suites.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@rednaxelafx @gatorsmile Please, review this PR. I haven't run tests locally, let's wait for jenkins.

**Comment by dongjoon-hyun** [comment]
Thank you for making a swift action, @MaxGekk .

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala [hunk 1]
```diff
@@ -188,12 +167,12 @@ case class Like(str: Expression, pattern: Expression, escape: Expression)
...
+      nullSafeCodeGen(ctx, ev, (eval1, eval2) => {
         s"""
-          String $patternStr = $eval2.toString();
-          $patternClass $compiledPattern = $patternClass.compile(
-            $escapeFunc($patternStr, '$newEscapeChar'));
-          ${ev.value} = $compiledPattern.matcher($eval1.toString()).matches();
+          String $$rightStr = $eval2.toString();
+          $patternClass $$pattern = $patternClass.compile(
+            $escapeFunc($$rightStr, '$newEscapeChar'));
+          ${ev.value} = $$pattern.matcher($eval1.toString()).matches();
```
Comment (rednaxelafx) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala]: Is the `$$` double-dollar intended here?

**Review by rednaxelafx — COMMENTED** [review]
Thanks for the quick response, @MaxGekk !
This revert mostly LGTM except for one inline comment below:

**Review by HyukjinKwon — APPROVED** [review]
Looks fine given the discussion made in the linked PR.

**Review by maropu — APPROVED** [review]

**Comment by MaxGekk** [comment]
jenkins, retest this, please

**Review by rednaxelafx — APPROVED** [review]
LGTM after the update. Thanks @MaxGekk !

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala [hunk 2]
```diff
@@ -188,12 +167,12 @@ case class Like(str: Expression, pattern: Expression, escape: Expression)
...
+      nullSafeCodeGen(ctx, ev, (eval1, eval2) => {
         s"""
-          String $patternStr = $eval2.toString();
-          $patternClass $compiledPattern = $patternClass.compile(
-            $escapeFunc($patternStr, '$newEscapeChar'));
-          ${ev.value} = $compiledPattern.matcher($eval1.toString()).matches();
+          String $rightStr = $eval2.toString();
+          $patternClass $pattern = $patternClass.compile(
+            $escapeFunc($rightStr, '$newEscapeChar'));
+          ${ev.value} = $pattern.matcher($eval1.toString()).matches();
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala]: This part is a logical reverting.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you all
Merged to master.

**Comment by dongjoon-hyun** [comment]
Could you make a backport to `branch-3.0`, @MaxGekk ? There is a conflict.

**Comment by HyukjinKwon** [comment]
Actually that conflict is by me https://github.com/apache/spark/pull/27514. Let me handle it.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon !

**Pull request closed without merging**